### PR TITLE
Issue 2008: Close connection if wire version is incompatible.

### LINKED ISSUE #2008: Enforce wire protocol versions
**Problem description**
If a client connects to the server with too old or too new of a protocol version the connection should be dropped.

**Problem location**
AppendProcessor.

**Suggestions for an improvement**
Check for non-overlap in the version ranges.

### PR DESCRIPTION
**Pull request by tkaitchuck**
Signed-off-by: Tom Kaitchuck <tom.kaitchuck@emc.com>

**Change log description**
If a client connects with an incompatible wire protocol, close the connection after sending the supported versions back to it.

**Purpose of the change**
Fixes #2008 

**What the code does**
Closes the connection if the client is incompatible. 

**How to verify it**
Should currently have no effect are all protocol versions are compatible.

### CODE REVIEW

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java [hunk 1]
```diff
@@ -133,6 +133,10 @@ public AppendProcessor(StreamSegmentStore store, ServerConnection connection, Re
     public void hello(Hello hello) {
         log.info("Received hello from connection: {}", connection);
         connection.send(new Hello(WireCommands.WIRE_VERSION, WireCommands.OLDEST_COMPATABLE_VERSION));
+        if (hello.getLowVersion() > WireCommands.WIRE_VERSION || hello.getHighVersion() < WireCommands.OLDEST_COMPATABLE_VERSION) {
+            log.warn("Incompatable wire protocol versions {} from connection {}", hello, connection);
```
Comment (andreipaduroiu) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java]: "Incompatible"

File: shared/protocol/src/main/java/io/pravega/shared/protocol/netty/FailingReplyProcessor.java [hunk 2]
```diff
@@ -41,7 +41,11 @@
 
     @Override
     public void hello(Hello hello) {
-        log.info("Received hello: {}", hello);
+        if (hello.getLowVersion() > WireCommands.WIRE_VERSION || hello.getHighVersion() < WireCommands.OLDEST_COMPATABLE_VERSION) {
+            log.error("Incompatable wire protocol versions {}", hello);
```
Comment (andreipaduroiu) [inline shared/protocol/src/main/java/io/pravega/shared/protocol/netty/FailingReplyProcessor.java]: and here

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java [hunk 3]
```diff
@@ -132,7 +132,11 @@ public AppendProcessor(StreamSegmentStore store, ServerConnection connection, Re
     @Override
     public void hello(Hello hello) {
         log.info("Received hello from connection: {}", connection);
-        connection.send(new Hello(WireCommands.WIRE_VERSION, WireCommands.OLDEST_COMPATIBLE_VERSION));
+        connection.send(new Hello(WireCommands.WIRE_VERSION, WireCommands.OLDEST_COMPATABLE_VERSION));
+        if (hello.getLowVersion() > WireCommands.WIRE_VERSION || hello.getHighVersion() < WireCommands.OLDEST_COMPATABLE_VERSION) {
```
Comment (fpj) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java]: `COMPATABLE` -> `COMPATIBLE`

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java [hunk 4]
```diff
@@ -132,7 +132,11 @@ public AppendProcessor(StreamSegmentStore store, ServerConnection connection, Re
     @Override
     public void hello(Hello hello) {
         log.info("Received hello from connection: {}", connection);
-        connection.send(new Hello(WireCommands.WIRE_VERSION, WireCommands.OLDEST_COMPATIBLE_VERSION));
+        connection.send(new Hello(WireCommands.WIRE_VERSION, WireCommands.OLDEST_COMPATABLE_VERSION));
```
Comment (fpj) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java]: `COMPATABLE` -> `COMPATIBLE`

File: shared/protocol/src/main/java/io/pravega/shared/protocol/netty/FailingReplyProcessor.java [hunk 5]
```diff
@@ -42,7 +42,11 @@
 
     @Override
     public void hello(Hello hello) {
-        log.info("Received hello: {}", hello);
+        if (hello.getLowVersion() > WireCommands.WIRE_VERSION || hello.getHighVersion() < WireCommands.OLDEST_COMPATABLE_VERSION) {
```
Comment (fpj) [inline shared/protocol/src/main/java/io/pravega/shared/protocol/netty/FailingReplyProcessor.java]: `COMPATABLE` -> `COMPATIBLE`

**Pull request merged**